src/Event/PreWriteEvent.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+final class PreWriteEvent extends Event
+{
+    const NAME = Events::PRE_WRITE;
```

Comment:
This is a really weird thing to do. Did I miss the discussion on why this is done this way?

Reply:
You mean, defining a `NAME` constant in the classes?
I did it mainly for comfort purpose, actually: it makes more sense to me to use a constant attached to the concerned class, so I kept it after @dunglas asked me to create the separate `Events` class.
Do you want me to remove the `NAME` constants?

Reply:
I don't see the value of these constants. Just use e.g. `Events::PRE_WRITE` directly.